src/Profile.cpp

```diff
@@ -787,15 +787,13 @@ void Profile::IncrementCoursePlayCount( const Course* pCourse, const Trail* pTra
 void Profile::GetAllUsedHighScoreNames(std::set<RString>& names)
 {
 #define GET_NAMES_FROM_MAP(main_member, main_key_type, main_value_type, sub_member, sub_key_type, sub_value_type) \
-	for(std::map<main_key_type, main_value_type>::iterator main_entry= \
+	for(auto main_entry= \

```

Comment:
I approve the use of `auto`. I would have suggested combining some of those lines again, though.